Sources/Aperture/Aperture.swift

```diff
@@ -1,228 +1,846 @@
+		}
 
-		if let audioDevice = audioDevice {
-			if !audioDevice.hasMediaType(.audio) {
-				throw Error.invalidAudioDevice
+		public func pause() throws {
+			guard let recordingSession else {
+				throw Error.recorderNotStarted
 			}
 
-			let audioInput = try AVCaptureDeviceInput(device: audioDevice)
+			recordingSession.pause()
+			onPause?()
+		}
 
-			if session.canAddInput(audioInput) {
-				session.addInput(audioInput)
-			} else {
-				throw Error.couldNotAddMic
+		public func resume() async throws {
+			guard let recordingSession else {
+				throw Error.recorderNotStarted
 			}
+
+			await recordingSession.resume()
+			onResume?()
 		}
+	}
+}
 
-		if session.canAddInput(input) {
-			session.addInput(input)
-		} else {
-			throw Error.couldNotAddScreen
+extension Aperture {
+	internal final class RecordingSession: NSObject {
+		/// The stream object for capturing anything on displays
+		private var stream: SCStream?
+		private var isStreamRecording = false
+		/// The writer that will combine all the video/audio inputs into the destination file
+		private var assetWriter: AVAssetWriter?
+
+		/// The video stream
+		private var videoInput: AVAssetWriterInput?
+		/// The optional system audio stream
+		private var systemAudioInput: AVAssetWriterInput?
+		/// The optional microphone audio stream
+		private var microphoneInput: AVAssetWriterInput?
+
+		/// The data output stream for the microphone session
+		private var microphoneDataOutput: AVCaptureAudioDataOutput?
+		/// The data output stream for the external device session
+		private var externalDeviceDataOutput: AVCaptureAudioDataOutput?
+
+		/// The microphone capturing session for versions before macOS 15
+		private var microphoneCaptureSession: AVCaptureSession?
+		/// The external device capturing session
+		private var externalDeviceCaptureSession: AVCaptureSession?
+
+		/// Activity to prevent going to sleep
+		private var activity: Activity?
+
+		/// Whether the recorder has started writting to the output file
+		private var isRunning = false
+		/// Whether the recorder is paused
+		private(set) var isPaused = false
+
+		/// Internal helpers for when we are resuming, used to fix the buffer timing
+		private var isResuming = false
+		private var timeOffset = CMTime.zero
+		private var lastFrame: CMTime?
+
+		/// Continuation to resolve when we have started writting to the output file
+		private var continuation: CheckedContinuation<Void, any Swift.Error>?
+		/// Continuation to resolve when the recording has resumed
+		private var resumeContinuation: CheckedContinuation<Void, Never>?
+		/// Holds any errors that are throwing during the recording, we throw them when we stop
+		private var error: Error?
+
+		/// The options for the current recording
+		private var options: RecordingOptions?
+		/// The target type for the current recording
+		private var target: Target?
+
+		/// The error handler for the recording session
+		var onError: ((Error) -> Void)?
+
+		func startRecording(
```

Comment:
```suggestion
		func start(
```

It's inconsistent with `pause` and the recorder part is already clear from the namespace. `recorder.start()`